### ISSUE TITLE
Right click to minimize all windows except active window

### ISSUE THREAD

**Issue by meybonomme** [amivaleo/Show-Desktop-Button#44]
I have often situation, when I have already too many windows open and I want to minimize all **except** active window to better focusing.

Best place to integrate this function is Show Desktop Button. Left click to button acts as now, but right click will minimize all windows except active window.

What do You think about this idea?

EDIT: I think GNOME top panel won't do difference, is it left, middle or right click. Am I right?

**Comment by amivaleo** [comment]
I have... Contrasting feelings about this.

On one hand, I feel that such feature would be much appreciated, but on the other...

* it would make the indicator on the panel work in different ways based on which "click" you gave (right-click vs left-click). This alone is a really no-no, in my opinion.
* I see them as two very different functions that maybe shouldn't belong to the same extension. Would you look for a "show desktop button" when you're actually looking for a "stay focused" extension? 🤔

The very purpose of this extension is to, as it says, to let you see the destkop by removing everything on top of it.

By itself, this extension shouldn't even exist, since vanilla gnome puts nothing on the desktop. Hence, there's no need to see the desktop since it's empty. Nothing's there apart from a wallpaper.

This extension makes some sense if you use desktop icons or conky or any other software that fills the emptyness on the desktop.

I tend to consider this request of yours as a request for a new extension, which I might be able to make (I'm not that much skilled, not at all) that would put another indicator on the panel that would exactly what you say. Actually, it's hard to believe it doesn't exist already. 🤔 Did you checked?

**Comment by meybonomme** [comment]
I got Your point. That was just an idea, but not all ideas are always the best.

>By itself, this extension shouldn't even exist, since vanilla gnome puts nothing on the desktop. Hence, there's no need to see the desktop since it's empty. Nothing's there apart from a wallpaper.

I'm using show desktup button extension when I have too many windows in screen, but I don't want to close them. Then I have "clean" desktop and I will unminimize only needed window at this moment. Sometimes I will just click to the show desktop button, to think and watch beautiful wallpaper slideshow pictures in my desktop.

My Fedora desktop is nice and clean, I don't use desktop icons or any widgets. These things are so yesterday and won't help even productivity in any way. Even in Windows I prefer clean desktop without icons.

I don't know how people can use their computers with desktop full icons and trash :laughing: 
That's why I love Gnome - it's so modern looking, flat, clean, minimalistic and very nice. I just prefer some things to set my way, not to Gnome way. Fedora is my project and second OS right now (although within last few weeks it's more like mine OS already :laughing: ) and Dash to Dock is "must be" for me. Without this extension Gnome is useless to me.

**Comment by amivaleo** [comment]
> That was just an idea, but not all ideas are always the best.

The idea is interesting, I'm not against that. The thing is that I believe it should be another extension. 🤔

> I'm using show desktup button extension when I have too many windows in screen, but I don't want to close them. Then I have "clean" desktop and I will unminimize only needed window at this moment.

I believe gnome "philosophy" is that, in cases like this, you should use workspaces. I'm not saying that this is my suggestion, I'm saying that maybe I should keep this in mind so to understand how a gnome user could be, who is my "target" when coding extensions.

**Comment by amivaleo** [comment]
After sooooo much time (sorry), I decided to try creating an extension based on tyour idea.
I've set up a repo [here](https://github.com/amivaleo/Hide-Unfocused).
It's not working yet, though.

Sorry for loooong wait.

**Comment by meybonomme** [comment]
Awesome! But you have already ideas how to get it working?

**Comment by amivaleo** [comment]
> Awesome! But you have already ideas how to get it working?

Well, using a shortcut. No panel indicator. 🤔
It will be a very light extension.
I wonder if there's another way to have the same feature without an extension... 🤔

Anyway, I will work on that. 💪

**Comment by meybonomme** [comment]
Do I understand correctly that you plan to make it a separate extension that works as an additional extension to the Show Desktop Button extension?

**Comment by amivaleo** [comment]
It is a separate extension. They won't "talk" to each other. They will be independent.

**Comment by meybonomme** [comment]
And it will be work only with keyboard shortcut?

Unfortunately, I'm not a very active user of keyboard shortcuts, I mostly prefer to use the mouse. The average computer user at least knows what Ctrl+C or Ctrl+V does, but rarely, if ever, uses these combinations. At the same time, this hiding of all windows except for the one in focus would help the average user a lot.

Separate extension and keyboard shortcut is nice, but for me all this concept makes sense only as part of Show Desktop Button extension. This can be an optional setting and doesn't even need to be on by default. But definitely this feature should be controllable also via the Show Desktop Button icon and with the mouse. From the user's point of view, these two actions are very similar - primary click on an icon hides/unhides all windows, secondary click on an icon hides all windows except the one in focus.

**Comment by amivaleo** [comment]
I am sorry Meybo, but I have a very different perspective... :\

I am completely and strongly against the idea to make any indicator on the panel triggering a different action when right-clicked. In my view, this goes entirely against what an indicator should be. I would strongly recommend that developers avoid adding any right-click functionality.

I understand that having this feature might help improve your workflow (and not just yours), which is why I decided to create a separate extension to meet this need. However, I strongly believe that 'hide unfocused' is a different functionality from what the 'show desktop button' provides: the former centers on focusing on a specific window, while the latter is aimed at focusing on the desktop and decluttering the user's view.

An indicator that can be enabled for 'Hide Unfocused' could be a compromise I’m willing to accept. However, I truly can't accept any right-click functionality; that goes against my view of how panel icons should work.

**Comment by meybonomme** [comment]
> I am sorry Meybo, but I have a very different perspective... :\

It's okay, I accept your point of view.

> I am completely and strongly against the idea to having any indicator on the panel trigger a different action when right-clicked. In my view, this goes entirely against what an indicator should be. I would strongly recommend that developers avoid adding any right-click functionality.

By the way, the most popular GNOME color picker extension uses different functions for primary click and secondary click: https://extensions.gnome.org/extension/3396/color-picker/

Primary click in indicator launch color picker and secondary click opens history and settings menu. Works very well and is completely logical to use.

But I accept that you don't like the concept and we can't agree on it :)

**Comment by amivaleo** [comment]
> the most popular GNOME color picker extension uses different functions for primary click and secondary click: https://extensions.gnome.org/extension/3396/color-picker/

I don't have the authority to tell that dev that he shouldn't do that, but I still firmly believe it’s not a good idea. IAlso because I see no point in having that submenu at all. 🤔

I want GNOME to be "touch-friendly." If other developers choose to use the right-click option, I would just blame them for it. 😶

I realize this might sound impolite, and I apologize for that, but I would feel like I’m breaking some unwritten rule if I added any right-click functionality.

Anyway, I promised you an extension that will do exactly what you asked for, apart from implementing it by adding another indicator (or just a shortcut) instead of using the one from the show-desktop-button. I hope this is still a good deal! ^_^
